graphene_sqlalchemy/tests/test_sort_enums.py

```diff
@@ -0,0 +1,394 @@
+            only_fields = ["id", "pet_kind"]
+
+    sort_arg = PetType.sort_argument()
+    sort_enum = sort_arg.type._of_type
+    assert list(sort_enum._meta.enum.__members__) == [
+        "ID_ASC",
+        "ID_DESC",
+        "PET_KIND_ASC",
+        "PET_KIND_DESC",
+    ]
+    assert sort_arg.default_value == ["ID_ASC"]
+
+
+def test_sort_argument_for_multi_column_pk():
+    class MultiPkTestModel(Base):
+        __tablename__ = "multi_pk_test_table"
+        foo = sa.Column(sa.Integer, primary_key=True)
+        bar = sa.Column(sa.Integer, primary_key=True)
+
+    class MultiPkTestType(SQLAlchemyObjectType):
+        class Meta:
+            model = MultiPkTestModel
+
+    sort_arg = MultiPkTestType.sort_argument()
+    assert sort_arg.default_value == ["FOO_ASC", "BAR_ASC"]
+
+
+def test_sort_argument_only_indexed():
+    class IndexedTestModel(Base):
+        __tablename__ = "indexed_test_table"
+        id = sa.Column(sa.Integer, primary_key=True)
+        foo = sa.Column(sa.Integer, index=False)
+        bar = sa.Column(sa.Integer, index=True)
+
+    class IndexedTestType(SQLAlchemyObjectType):
+        class Meta:
+            model = IndexedTestModel
+
+    sort_arg = IndexedTestType.sort_argument(only_indexed=True)
+    sort_enum = sort_arg.type._of_type
+    assert list(sort_enum._meta.enum.__members__) == [
+        "ID_ASC",
+        "ID_DESC",
+        "BAR_ASC",
+        "BAR_DESC",
+    ]
+    assert sort_arg.default_value == ["ID_ASC"]
+
+
+def test_sort_argument_with_custom_symbol_names():
+    class PetType(SQLAlchemyObjectType):
+        class Meta:
+            model = Pet
+
+    def get_symbol_name(column_name, sort_asc=True):
+        return to_type_name(column_name) + ("Up" if sort_asc else "Down")
+
+    sort_arg = PetType.sort_argument(get_symbol_name=get_symbol_name)
+    sort_enum = sort_arg.type._of_type
+    assert list(sort_enum._meta.enum.__members__) == [
+        "IdUp",
+        "IdDown",
+        "NameUp",
+        "NameDown",
+        "PetKindUp",
+        "PetKindDown",
+        "HairKindUp",
+        "HairKindDown",
+        "ReporterIdUp",
+        "ReporterIdDown",
+    ]
+    assert sort_arg.default_value == ["IdUp"]
+
+
+def test_sort_query(session):
+    add_pets(session)
+
+    class PetNode(SQLAlchemyObjectType):
+        class Meta:
+            model = Pet
+            interfaces = (Node,)
+
+    class PetConnection(Connection):
+        class Meta:
+            node = PetNode
+
+    class Query(ObjectType):
+        defaultSort = SQLAlchemyConnectionField(PetConnection)
+        nameSort = SQLAlchemyConnectionField(PetConnection)
+        multipleSort = SQLAlchemyConnectionField(PetConnection)
+        descSort = SQLAlchemyConnectionField(PetConnection)
+        singleColumnSort = SQLAlchemyConnectionField(
+            PetConnection, sort=Argument(PetNode.sort_enum())
+        )
+        noDefaultSort = SQLAlchemyConnectionField(
+            PetConnection, sort=PetNode.sort_argument(False)
```

Comment:
Yes, you're right. And it reveals that the NoDefaultSort tests was actually not working.

I noticed that it uses `set()` when doing the comparison, so it never really checked the order. In fact, it's not that simple to check whether no sorting happens since sqlite does not guarantee any sort order with no "order by". Usually it's the order of the primary key index, which is exactly the same as the default order. We would need to contrive something different and more complicated to check that it's really not sorting. I suggest creating a separate issue for that, though, to not delay this any further.